nemo/core/classes/modelPT.py

```diff
@@ -112,22 +113,22 @@ def setup_optimization(self, optim_config: Optional[Union[DictConfig, Dict]] = N
                 kwargs will be built and supplied to instantiate the optimizer.
         """
         # Setup optimizer and scheduler
-        if 'sched' in optim_config and self._cfg is not None and 'trainer' in self._cfg.pl:
-            if self._cfg.pl.trainer.max_steps is None:
-                if self._cfg.pl.trainer.gpus == 0:
+        if 'sched' in optim_config and self._trainer is not None:
+            if not isinstance(self._trainer.accumulate_grad_batches, int):
+                raise ValueError("We do not currently support gradient acculumation that is not an integer.")
+            if self._trainer.max_steps is None:
+                if self._trainer.gpus == 0:
                     # training on CPU
-                    iters_per_batch = self._cfg.pl.trainer.max_epochs / float(
-                        self._cfg.pl.trainer.num_nodes * self._cfg.pl.trainer.accumulate_grad_batches
+                    iters_per_batch = self._trainer.max_epochs / float(
+                        self._trainer.num_nodes * self._trainer.accumulate_grad_batches
```

Comment:
Note that now we can just grab the relevant information from the lightning trainer object as opposed to interpreting command line arugments